### ISSUE TITLE
WebContentView container priority is too high

### ISSUE THREAD

**Issue by Y2zz** [electron/electron#44867]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Adding WebContentView to BrowseWindow will obscure the render thread UI elements, and positioning cannot solve this problem

### Proposed Solution

Refer to the comparison with webview, you can control the hierarchical relationship by yourself, or provide an api for control

### Alternatives Considered

none

### Additional Information

Demo: https://gist.github.com/7779d612dd7cdf8482f91dbfee266767

**Comment by nikwen** [comment]
If you want to change the z-level of the `WebContentsView`, you might be able to switch to a `BaseWindow` with two `WebContentsView`s instead of a `BrowserWindow` with one additional `WebContentsView`.

(That said, I'm not entirely sure if that's what you're looking to accomplish. If it's not, it would be great if you could explain in more detail what you're looking for.)

**Comment by nikwen** [comment]
Yes, you can just use a `BaseWindow` with multiple `WebContentsView`s added to it. One for the tabs and one for the actual web content.

That should be possible without any changes to Electron.

**Comment by nikwen** [comment]
True. You could create another `WebContentsView` for the popup and put it on top of everything.

In general, if you're building a browser, you might be better off using Chromium directly instead of using Electron.

**Comment by Jonovono** [comment]
@Y2zz What did you end up doing? I am also trying to decide if I should use WebContentsView or just keep using webview despite the recommendation to not use it.

**Comment by Jonovono** [comment]
> > [@Y2zz](https://github.com/Y2zz) What did you end up doing? I am also trying to decide if I should use WebContentsView or just keep using webview despite the recommendation to not use it.
> 
> Do your needs match mine? I finally used `Webview` to implement it

So you just use the <webview> tag? Do you have any performance issues with lots of tabs? That's my main concern, but it's much easier to work with for sure. I have similar goal as you

**Comment by zoy-l** [comment]
hack
BaseWindow + main WebContentView, set to full window, with the bottom set to transparent. Change the z-order when the mouse hovers over the tab. 😈

**Comment by zmcode** [comment]
Your WebContentsView is overwritten, you can cache it, you can add it again, it's the top level

![Image](https://github.com/user-attachments/assets/57929b95-aa2e-4ba0-a387-8ac67fdb2e0b)

**Comment by codebytere** [comment]
Hi, while we appreciate this enhancement request it's not something that our team will be able to implement in the near future, and since we use our issues tab as the team's backlog, we prefer to only retain issues that are on our immediate roadmap. If you'd like to make a PR for this request we would welcome that but in the meantime I'm going to go ahead and close this.

**Comment by zmcode** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
